dateutil/rrule.py

```diff
@@ -1563,8 +1570,22 @@ def _parse_rfc(self, s,
                     # RFC 5445 3.8.2.4: The VALUE parameter is optional, but
                     # may be found only once.
                     value_found = False
+                    TZID = None
                     valid_values = {"VALUE=DATE-TIME", "VALUE=DATE"}
                     for parm in parms:
+                        if parm.startswith("TZID="):
+                            try:
+                                tzkey = TZID_NAMES[parm.split('TZID=')[-1]]
+                            except KeyError:
+                                continue
+                            if callable(tzinfos):
+                                tzlookup = tzinfos
+                            elif isinstance(tzinfos, dict):
+                                tzlookup = tzinfos.get
+                            else:
```

Comment:
Once again I don't think this is the right thing to do. `tzlookup` should be set to `tz.gettz` if and only if the user requested the *default* behavior.

I would probably do it this way:

```python
if tzinfos is None:
    tzlookup = tz.gettz
elif callable(tzinfos):
    tzlookup = tzinfos
else:
    tzlookup = getattr(tzinfos, 'get', None)
    if tzlookup is None:
        raise ValueError('tzinfos must be a callable or mapping')
```